CMakeLists.txt

```diff
@@ -0,0 +1,7 @@
+cmake_minimum_required(VERSION 2.8)
+
+add_subdirectory(src/)
+option(COMPILE_TEST "Compile Tests" ON)
+if(COMPILE_TEST) 
```

Comment:
There is a general purpose CMake option called BUILD_TESTING ( https://cmake.org/cmake/help/v3.1/module/CTest.html ).